platforms/native/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/xcode/MacOSSdkPlatformPathLocator.java

```diff
@@ -30,6 +30,6 @@ public MacOSSdkPlatformPathLocator(ExecActionFactory execActionFactory) {
 
     @Override
     protected List<String> getXcrunFlags() {
-        return ImmutableList.of("--show-sdk-platform-path");
+        return ImmutableList.of("--show-sdk-path");
```

Comment:
This class is used by the XCTest plugin to locate XCTest. However, I don't think the SDK contains XCTest. XCTest lives in the platform (which also contains the SDK).

So, there should be some related change to the XCTest plugin to allow it to continue to locate XCTest.

There are a couple of options:

1. Still run `xcrun --show-sdk-platform-path` and disable XCTest support if it fails (and tell the user they will need the full XCode install)
2. Use `--show-sdk-path` and traverse its parent directories to locate XCTest (ie assume a particular layout). And also maybe let the user know if it's missing.

For (2) `MacOSSdkPlatformPathLocator` should be removed rather than modified, as it would do the same thing as `MacOSSdkPathLocator`.

I'd probably do option (1).

Reply:
If `xcrun --show-sdk-platform-path` throws on my machine, that means that I have XCTest and that I should uninstall it?

Reply:
It's the other way around. If it throws, then you don't have XCTest installed and you should either install XCTest (by installing XCode) if you actually need it, or remove the xctest plugin if you're not using XCTest.

Reply:
XCode is installed and it fails like this. I have a hard time making sense out of it.
![image](https://github.com/user-attachments/assets/87c5b24d-0d6a-454b-bc38-374fd0798f81)

Reply:
Hey @oldergod,

What happens if you run `xcrun --sdk macosx --show-sdk-platform-path`?

Reply:
Thank you. It's been a while and can't remember things. I'll open again if I encounter new problems after Swift6 support is out. Thank you